### PR TITLE
refactor: tidy up some old code

### PR DESCRIPTION
**Pull request by elianiva**
- use pnpm instead of npm
- changed indent style from 2 to 4 to make it look less cluttered
- extract prop typings into its own separate type definition
- dedupe type definition
- remove props destructuring since it makes it less clear if a value comes from props
- fix hydration mismatch when rendering caused by figure nested inside p
- safer null/undefined check by enforcing the value to be boolean
- added missing type annotations in some components
- updated svgr config according to docs

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
+        name: Build
+        runs-on: ubuntu-latest
+        timeout-minutes: 15
 
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v2
```
Comment (aldy505) [inline .github/workflows/ci.yml:17]:
```suggestion
              uses: actions/checkout@v3
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
+              uses: actions/setup-node@v2.1.5
+              with:
+                  node-version: 16.x
 
-      - name: Installling dependencies
-        if: steps.cache.outputs.cache-hit != 'true'
-        run: npm install
+            - name: Installling dependencies
+              if: steps.cache.outputs.cache-hit != 'true'
+              run: npm install
```
Comment (aldy505) [inline .github/workflows/ci.yml:26]: ini ga sekalian diubah ke pnpm?

File: package.json [hunk 3]
```diff
@@ -1,64 +1,64 @@
...
+    },
+    "dependencies": {
+        "@fontsource/jetbrains-mono": "5.0.3",
+        "@fontsource/lusitana": "5.0.3",
+        "@fontsource/rubik": "^5.0.3",
+        "@giscus/react": "2.2.8",
+        "@svgr/webpack": "^8.0.1",
+        "clsx": "^1.2.1",
+        "gray-matter": "4.0.3",
+        "next": "^13.4.5",
```
Comment (aldy505) [inline package.json:31]: mantap

File: package.json [hunk 4]
```diff
@@ -1,64 +1,64 @@
...
+        "react": "18.2.0",
+        "react-dom": "18.2.0",
+        "rehype-katex": "6.0.3",
+        "rehype-pretty-code": "^0.9.7",
+        "rehype-raw": "^6.1.1",
+        "remark-math": "5.1.1",
+        "remark-parse": "10.0.2",
+        "remark-unwrap-images": "^3.0.1",
+        "sharp": "0.32.1",
+        "shiki": "0.14.2",
```
Comment (aldy505) [inline package.json:44]: kalo dua ini di bump, polarite harus di urus juga ini.

Comment (aldy505) [reply package.json:44]: weekend depan deh gw sentuh

File: utils/transformMdx.ts [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
+import rehypePrettyCode, { type Options } from "rehype-pretty-code";
+import remarkMath from "remark-math";
+import remarkUnwrapImages from "remark-unwrap-images";
 
-const codeHighlightOption: Partial<Options> = {
-  theme: 'github-dark',
-};
+const codeHighlightOption = {
+    theme: "github-dark",
+} satisfies Partial<Options>;
```
Comment (aldy505) [inline utils/transformMdx.ts:10]: anjay

**Review by aldy505 — APPROVED** [review]

File: components/ImageWithFrame.tsx [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
-export function ImageWithFrame({ src, alt }: DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>) {
+type ImageWithFrameProps = DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>;
+
+export function ImageWithFrame(props: ImageWithFrameProps) {
+    if (props.src === undefined) return null;
     return (
         <figure className="text-center">
-            <img className="inline-block p-0 overflow-hidden rounded-md" src={src} alt={alt} />
-            <figcaption className="break-words">{alt}</figcaption>
+            <img className="inline-block p-0 overflow-hidden rounded-md" src={props.src} alt={props.alt ?? "-"} />
```
Comment (krowter) [inline components/ImageWithFrame.tsx:9]: fallback buat `alt`nya empty string aja mending
https://www.w3.org/WAI/tutorials/images/decorative/

File: components/ReadAnyway.tsx [hunk 7]
```diff
@@ -27,10 +27,11 @@ export function ReadAnyway({ posts }: ReadAnywayProps) {
...
 
-                <Link href="/blog">
-                    <a className="w-full font-bold text-black dark:text-white text-center block mt-10 hover:text-primary-600 dark:hover:text-primary-200 transition duration-300">
-                        FIND MORE POST
-                    </a>
+                <Link
+                    href="/blog"
+                    className="w-full font-bold text-black dark:text-white text-center block mt-10 hover:text-primary-600 dark:hover:text-primary-200 transition duration-300"
+                >
+                    FIND MORE POST
```
Comment (krowter) [inline components/ReadAnyway.tsx:34]: Mintol casing nya diubah kayak biasa, terus className `uppercase`, biar ngga dieja satu2 sama screenreader

File: package.json (hunk 4 above)
Comment (aldy505) [reply package.json:44]: KOK POLARITE. harusnya graphene.

**Comment by aldy505** [comment]
Should bump to Node 20

**Comment by aldy505** [comment]
Some issues with this PR:

1. GitHub actions workflow error:
    [Invalid workflow file: .github/workflows/ci.yml#L38](https://github.com/teknologi-umum/blog/actions/runs/5301580643/workflow)
    The workflow is not valid. .github/workflows/ci.yml (Line: 38, Col: 20): Unexpected value '' .github/workflows/ci.yml (Line: 39, Col: 15): Unexpected value 'path'
2. Netlify build error: (see https://app.netlify.com/sites/teknum/deploys/648e7a802b726c00074b6a53)
```
10:31:54 AM: Netlify Build                                                 
10:31:54 AM: ────────────────────────────────────────────────────────────────
10:31:54 AM: ​
10:31:54 AM: ❯ Version
10:31:54 AM:   @netlify/build 29.12.1
10:31:54 AM: ​
10:31:54 AM: ❯ Flags
10:31:54 AM:   baseRelDir: true
10:31:54 AM:   buildId: 648e7a802b726c00074b6a51
10:31:54 AM:   deployId: 648e7a802b726c00074b6a53
10:31:54 AM: ​
10:31:54 AM: ❯ Current directory
10:31:54 AM:   /opt/build/repo
10:31:54 AM: ​
10:31:54 AM: ❯ Config file
10:31:54 AM:   /opt/build/repo/netlify.toml
10:31:54 AM: ​
10:31:54 AM: ❯ Context
10:31:54 AM:   deploy-preview
10:31:54 AM: ​
10:31:54 AM: ❯ Using Next.js Runtime - v3.9.2
10:31:54 AM: ​
10:31:54 AM: ❯ Outdated plugins
10:31:54 AM:    - @netlify/plugin-nextjs@3.9.2: latest version is 4.37.4
10:31:54 AM:      To upgrade this plugin, please uninstall and re-install it from the Netlify plugins directory (https://app.netlify.com/plugins)
10:31:55 AM: ​
10:31:55 AM: @netlify/plugin-nextjs (onPreBuild event)                     
10:31:55 AM: ────────────────────────────────────────────────────────────────
10:31:55 AM: ​
10:31:55 AM: Using Next.js 13.4.6
10:31:55 AM: Warning: support for Next.js >=11.1.0 is experimental
10:31:55 AM: NOTE: Static HTML export Next.js projects (projects that use `next export`) do not require most of this plugin. For these sites, this plugin *only* caches builds.
10:31:55 AM: ​
10:31:55 AM: (@netlify/plugin-nextjs onPreBuild completed in 9ms)
10:31:55 AM: ​
10:31:55 AM: build.command from netlify.toml                               
10:31:55 AM: ────────────────────────────────────────────────────────────────
10:31:55 AM: ​
10:31:55 AM: $ npm run build && npm run export
10:31:55 AM: > teknum-blog@0.1.0 build
10:31:55 AM: > next build
10:31:56 AM: - warn No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
10:31:56 AM: - info Linting and checking validity of types...
10:31:58 AM: - info Creating an optimized production build...
10:32:10 AM: Failed to compile.
10:32:10 AM: 
10:32:10 AM: ./services/post-by-slug.ts
10:32:10 AM: Module not found: Can't resolve 'fs/promises'
10:32:10 AM: https://nextjs.org/docs/messages/module-not-found
10:32:10 AM: Import trace for requested module:
10:32:10 AM: ./services/index.ts
10:32:10 AM: ./pages/search.tsx
10:32:10 AM: ./services/post-slugs.ts
10:32:10 AM: Module not found: Can't resolve 'fs/promises'
10:32:10 AM: https://nextjs.org/docs/messages/module-not-found
10:32:10 AM: Import trace for requested module:
10:32:10 AM: ./services/index.ts
10:32:10 AM: ./pages/search.tsx
10:32:10 AM: > Build failed because of webpack errors
10:32:11 AM: ​
10:32:11 AM: "build.command" failed                                        
10:32:11 AM: ────────────────────────────────────────────────────────────────
10:32:11 AM: ​
10:32:11 AM:   Error message
10:32:11 AM:   Command failed with exit code 1: npm run build && npm run export (https://ntl.fyi/exit-code-1)
10:32:11 AM: ​
10:32:11 AM:   Error location
10:32:11 AM:   In build.command from netlify.toml:
10:32:11 AM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
10:32:11 AM:   npm run build && npm run export
10:32:11 AM: ​
10:32:11 AM:   Resolved config
10:32:11 AM:   build:
10:32:11 AM:     base: /opt/build/repo
10:32:11 AM:     command: npm run build && npm run export
10:32:11 AM:     commandOrigin: config
10:32:11 AM:     environment:
10:32:11 AM:       - REVIEW_ID
10:32:11 AM:     publish: /opt/build/repo/out
10:32:11 AM:     publishOrigin: config
10:32:11 AM:   plugins:
10:32:11 AM:     - inputs: {}
10:32:11 AM:       origin: ui
10:32:11 AM:       package: '@netlify/plugin-nextjs'
10:32:11 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
10:32:11 AM: Failing build: Failed to build site
10:32:12 AM: Finished processing build request in 58.313s
```

**Comment by aldy505** [comment]
@elianiva the build still failed

**Comment by elianiva** [comment]
ga ngerti apa yg salah sama netlify, di local bisa, mau cek ga punya akses netlify

**Comment by aldy505** [comment]
minta akses ke @lamualfa dia yang punya akun kalo gasalah

**Comment by lamualfa** [comment]
`@netlify/plugin-nextjs` harus diupdate dulu.

**Comment by elianiva** [comment]
> `@netlify/plugin-nextjs` harus diupdate dulu.

tapi kita ngga pake itu

**Comment by aldy505** [comment]
might be relevant: https://github.com/vercel/next.js/discussions/34992

I don't know for sure though, I don't code in next

**Comment by lamualfa** [comment]
Pluginnya udah gw update tapi masih error di `fs/promises`.

File: next.config.js [hunk 8]
```diff
@@ -1,24 +1,16 @@
...
-      use: [
-        'next-swc-loader',
-        {
-          loader: '@svgr/webpack',
-          options: { babel: false },
-        },
-      ],
-    });
-    config.resolve.fallback = {
-      fs: false,
```
Comment (lamualfa) [inline next.config.js:18]: Ini dibalikin lagi biar nggak error.

**Review by lamualfa — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 8 above)
Comment (aldy505) [reply next.config.js:18]: @elianiva ikuti perintah lord kita

**Review by lamualfa — CHANGES_REQUESTED** [review]
https://github.com/teknologi-umum/blog/blob/7860e27cb6a24648cee967d69fd81c54016ee604/netlify.toml#L3

Publish directorynya dihapus biar nggak error.

**Review by lamualfa — APPROVED** [review]

**Pull request merged**